ledger/eval.go

```diff
@@ -1075,23 +1103,29 @@ func (validator *evalTxValidator) run() {
 
 // used by Ledger.Validate() Ledger.AddBlock() Ledger.trackerEvalVerified()(accountUpdates.loadFromDisk())
 //
-// Validate: eval(ctx, l, blk, true, txcache, executionPool)
-// AddBlock: eval(context.Background(), l, blk, false, txcache, nil)
-// tracker:  eval(context.Background(), l, blk, false, txcache, nil)
-func eval(ctx context.Context, l ledgerForEvaluator, blk bookkeeping.Block, validate bool, txcache verify.VerifiedTransactionCache, executionPool execpool.BacklogPool) (ledgercore.StateDelta, error) {
+// Validate: eval(ctx, l, blk, true, txcache, executionPool, true)
+// AddBlock: eval(context.Background(), l, blk, false, txcache, nil, true)
+// tracker:  eval(context.Background(), l, blk, false, txcache, nil, false)
+func eval(ctx context.Context, l ledgerForEvaluator, blk bookkeeping.Block, validate bool, txcache verify.VerifiedTransactionCache, executionPool execpool.BacklogPool, usePrefetch bool) (ledgercore.StateDelta, error) {
 	eval, err := startEvaluator(l, blk.BlockHeader, len(blk.Payset), validate, false)
 	if err != nil {
 		return ledgercore.StateDelta{}, err
 	}
 
+	validationCtx, validationCancel := context.WithCancel(ctx)
+	defer validationCancel()
+
+	// If validationCtx or underlying ctx are Done, end prefetch
+	if usePrefetch {
+		go prefetchThread(validationCtx, eval.state.lookupParent, blk.Payset)
```

Comment:
why `eval.state.lookupParent` and not `eval.state` ?
`lookupParent` is a ledger the goes directly into accoutdb without any intermediate caching. Or does this relies on `au.baseAccounts` cache?
If instead `eval.state` is used then all these accounts are 100% cached within `roundCowState.mods`.

Reply:
`eval.state.lookupParent` is a roundCowBase with caching
`eval.state` is just a roundCowState.
I figure I may as well go straight to the layer with the caching.